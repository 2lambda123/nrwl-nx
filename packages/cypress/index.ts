export { cypressProjectGenerator } from './src/generators/cypress-project/cypress-project';
export { cypressInitGenerator } from './src/generators/init/init';
export { conversionGenerator } from './src/generators/convert-tslint-to-eslint/convert-tslint-to-eslint';
export { cypressComponentProject } from './src/generators/cypress-component-project/cypress-component-project';
export { migrateCypressProject } from './src/generators/migrate-to-cypress-11/migrate-to-cypress-11';
export { cypressE2EConfigurationGenerator } from './src/generators/cypress-e2e-configuration/cypress-e2e-configuration';
