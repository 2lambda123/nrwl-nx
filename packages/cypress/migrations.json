{
  "schematics": {
    "update-8.1.0": {
      "version": "8.1.0",
      "description": "Update cypress",
      "factory": "./migrations/update-8-1-0/update-8-1-0"
    }
  }
}
