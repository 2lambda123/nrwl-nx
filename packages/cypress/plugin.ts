export { createNodes, NxCypressMetadata } from './src/plugins/plugin';
