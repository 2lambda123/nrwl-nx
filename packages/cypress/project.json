{
  "name": "cypress",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/cypress",
  "projectType": "library",
  "targets": {
    "test": {},
    "build-base": {
      "executor": "@nrwl/js:tsc",
      "options": {
        "outputPath": "build/packages/cypress",
        "tsConfig": "packages/cypress/tsconfig.lib.json",
        "main": "packages/cypress/index.ts",
        "updateBuildableProjectDepsInPackageJson": false,
        "assets": [
          {
            "input": "packages/cypress",
            "glob": "**/files/**",
            "output": "/"
          },
          {
            "input": "packages/cypress",
            "glob": "**/files/**/.gitkeep",
            "output": "/"
          },
          {
            "input": "packages/cypress",
            "glob": "**/*.json",
            "ignore": ["**/tsconfig*.json", "project.json", ".eslintrc.json"],
            "output": "/"
          },
          {
            "input": "packages/cypress",
            "glob": "**/*.js",
            "ignore": ["**/jest.config.js"],
            "output": "/"
          },
          {
            "input": "packages/cypress",
            "glob": "**/*.d.ts",
            "output": "/"
          },
          {
            "input": "",
            "glob": "LICENSE",
            "output": "/"
          }
        ]
      },
      "outputs": ["{options.outputPath}"]
    },
    "build": {
      "executor": "nx:run-commands",
      "outputs": ["{workspaceRoot}/build/packages/cypress"],
      "options": {
        "command": "node ./scripts/copy-readme.js cypress"
      }
    },
    "lint": {}
  }
}
