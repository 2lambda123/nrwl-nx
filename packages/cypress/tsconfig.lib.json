{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "module": "commonjs",
    "outDir": "../../dist/out-tsc",
    "declaration": true,
    "types": ["node"]
  },
  "exclude": ["**/*.spec.ts", "**/*.test.ts", "jest.config.ts"],
  "include": [
    "**/*.ts",
    "src/plugins/preprocessor-vite.mts",
    "plugins/cypress-preset.mts"
  ]
}
