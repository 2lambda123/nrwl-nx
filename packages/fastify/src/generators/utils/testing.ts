import { Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

export function createTreeWithFastifyApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace();
  tree.write(
    'workspace.json',
    String.raw`
      {
        "projects": {
          "${appName}": {
            "root": "apps/api",
            "sourceRoot": "apps/api/src",
            "projectType": "application",
            "prefix": "api",
            "targets":{}
          }
        }
      }
    `
  );

  return tree;
}
