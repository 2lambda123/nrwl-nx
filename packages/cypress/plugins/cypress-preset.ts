import {
  createProjectGraphAsync,
  logger,
  parseTargetString,
  workspaceRoot,
} from '@nx/devkit';
import { dirname, join, relative } from 'path';
import { lstatSync } from 'fs';

import vitePreprocessor from '../src/plugins/preprocessor-vite';
import { ChildProcess, fork } from 'node:child_process';
import { createExecutorContext } from '../src/utils/ct-helpers';
import { startDevServer } from '../src/utils/start-dev-server';

interface BaseCypressPreset {
  videosFolder: string;
  screenshotsFolder: string;
  chromeWebSecurity: boolean;
}

export interface NxComponentTestingOptions {
  /**
   * the component testing target name.
   * this is only when customized away from the default value of `component-test`
   * @example 'component-test'
   */
  ctTargetName?: string;
  bundler?: 'vite' | 'webpack';
  compiler?: 'swc' | 'babel';
}

export function nxBaseCypressPreset(
  pathToConfig: string,
  options?: { testingType: 'component' | 'e2e' }
): BaseCypressPreset {
  // used to set babel settings for react CT.
  process.env.NX_CYPRESS_COMPONENT_TEST =
    options?.testingType === 'component' ? 'true' : 'false';
  // prevent from placing path outside the root of the workspace
  // if they pass in a file or directory
  const normalizedPath = lstatSync(pathToConfig).isDirectory()
    ? pathToConfig
    : dirname(pathToConfig);
  const projectPath = relative(workspaceRoot, normalizedPath);
  const offset = relative(normalizedPath, workspaceRoot);
  const videosFolder = join(offset, 'dist', 'cypress', projectPath, 'videos');
  const screenshotsFolder = join(
    offset,
    'dist',
    'cypress',
    projectPath,
    'screenshots'
  );

  return {
    videosFolder,
    screenshotsFolder,
    chromeWebSecurity: false,
  };
}

/**
 * nx E2E Preset for Cypress
 * @description
 * this preset contains the base configuration
 * for your e2e tests that nx recommends.
 * you can easily extend this within your cypress config via spreading the preset
 * @example
 * export default defineConfig({
 *   e2e: {
 *     ...nxE2EPreset(__dirname)
 *     // add your own config here
 *   }
 * })
 *
 */
export function nxE2EPreset(
  pathToConfig: string,
  options?: NxCypressE2EPresetOptions
) {
  const basePath = options?.cypressDir || 'src';
  const baseConfig: any /** Cypress.EndToEndConfigOptions */ = {
    ...nxBaseCypressPreset(pathToConfig),
    fileServerFolder: '.',
    supportFile: `${basePath}/support/e2e.ts`,
    specPattern: `${basePath}/**/*.cy.{js,jsx,ts,tsx}`,
    fixturesFolder: `${basePath}/fixtures`,
    env: {
      devServerTargets: options?.devServerTargets,
      devServerTargetOptions: {},
      ciDevServerTarget: options?.ciDevServerTarget,
    },
    async setupNodeEvents(on, config) {
      if (options?.bundler === 'vite') {
        on('file:preprocessor', vitePreprocessor());
      }
      if (!config.env.devServerTargets) {
        return;
      }
      const devServerTarget =
        config.env.devServerTarget ?? config.env.devServerTargets['default'];

      if (!devServerTarget) {
        return;
      }
      if (!config.baseUrl && devServerTarget) {
        const graph = await createProjectGraphAsync();
        const target = parseTargetString(devServerTarget, graph);
        const context = createExecutorContext(
          graph,
          graph.nodes[target.project].data?.targets,
          target.project,
          target.target,
          target.configuration
        );

        const devServer = startDevServer(
          {
            devServerTarget,
            ...config.env.devServerTargetOptions,
          },
          context
        );
        on('after:run', () => {
          devServer.return();
        });
        const devServerValue = (await devServer.next()).value;
        if (!devServerValue) {
          return;
        }
        return { ...config, baseUrl: devServerValue.baseUrl };
      }
    },
  };

  return baseConfig;
}

export type NxCypressE2EPresetOptions = {
  bundler?: string;
  /**
   * The directory from the project root, where the cypress files (support, fixtures, specs) are located.
   * i.e. 'cypress/' or 'src'
   * default is 'src'
   **/
  cypressDir?: string;

  devServerTargets?: Record<string, string>;
  ciDevServerTarget?: string;
};
