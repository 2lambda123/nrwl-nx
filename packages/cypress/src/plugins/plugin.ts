import {
  CreateDependencies,
  CreateNodes,
  CreateNodesContext,
  readJsonFile,
  TargetConfiguration,
  writeJsonFile,
} from '@nx/devkit';
import { dirname, extname, join } from 'path';
import { registerTsProject } from '@nx/js/src/internal';

import { getRootTsConfigPath } from '@nx/js';

import { CypressExecutorOptions } from '../executors/cypress/cypress.impl';
import { readTargetDefaultsForTarget } from 'nx/src/project-graph/utils/project-configuration-utils';
import { getNamedInputs } from '@nx/devkit/src/utils/get-named-inputs';
import { existsSync, readdirSync } from 'fs';
import { globWithWorkspaceContext } from 'nx/src/utils/workspace-context';
import { calculateHashForCreateNodes } from '@nx/devkit/src/utils/calculate-hash-for-create-nodes';
import { projectGraphCacheDirectory } from 'nx/src/utils/cache-directory';

export interface CypressPluginOptions {
  ciTargetName?: string;
  targetName?: string;
  componentTestingTargetName?: string;
}

const cachePath = join(projectGraphCacheDirectory, 'cypress.hash');
const targetsCache = existsSync(cachePath) ? readTargetsCache() : {};

const calculatedTargets: Record<
  string,
  Record<string, TargetConfiguration>
> = {};

function readTargetsCache(): Record<
  string,
  Record<string, TargetConfiguration<CypressExecutorOptions>>
> {
  return readJsonFile(cachePath);
}

function writeTargetsToCache(
  targets: Record<
    string,
    Record<string, TargetConfiguration<CypressExecutorOptions>>
  >
) {
  writeJsonFile(cachePath, targets);
}

export const createDependencies: CreateDependencies = () => {
  writeTargetsToCache(calculatedTargets);
  return [];
};

export const createNodes: CreateNodes<CypressPluginOptions> = [
  '**/cypress.config.{js,ts,mjs,mts,cjs,cts}',
  (configFilePath, options, context) => {
    options = normalizeOptions(options);
    const projectRoot = dirname(configFilePath);

    // Do not create a project if package.json and project.json isn't there.
    const siblingFiles = readdirSync(join(context.workspaceRoot, projectRoot));
    if (
      !siblingFiles.includes('package.json') &&
      !siblingFiles.includes('project.json')
    ) {
      return {};
    }

    const hash = calculateHashForCreateNodes(projectRoot, options, context);

    const targets = targetsCache[hash]
      ? targetsCache[hash]
      : buildCypressTargets(configFilePath, projectRoot, options, context);

    calculatedTargets[hash] = targets;

    return {
      projects: {
        [projectRoot]: {
          projectType: 'application',
          targets,
        },
      },
    };
  },
];

function getOutputs(
  projectRoot: string,
  cypressConfig: any,
  testingType: 'e2e' | 'component'
): string[] {
  function getOutput(path: string): string {
    if (path.startsWith('..')) {
      return join('{workspaceRoot}', join(projectRoot, path));
    } else {
      return join('{projectRoot}', path);
    }
  }

  const { screenshotsFolder, videosFolder, e2e, component } = cypressConfig;
  const outputs = ['{options.videosFolder}', '{options.screenshotsFolder}'];

  if (videosFolder) {
    outputs.push(getOutput(videosFolder));
  }

  if (screenshotsFolder) {
    outputs.push(getOutput(screenshotsFolder));
  }

  switch (testingType) {
    case 'e2e': {
      if (e2e.videosFolder) {
        outputs.push(getOutput(e2e.videosFolder));
      }
      if (e2e.screenshotsFolder) {
        outputs.push(getOutput(e2e.screenshotsFolder));
      }
      break;
    }
    case 'component': {
      if (component.videosFolder) {
        outputs.push(getOutput(component.videosFolder));
      }
      if (component.screenshotsFolder) {
        outputs.push(getOutput(component.screenshotsFolder));
      }
      break;
    }
  }

  return outputs;
}
function buildCypressTargets(
  configFilePath: string,
  projectRoot: string,
  options: CypressPluginOptions,
  context: CreateNodesContext
) {
  const cypressConfig = getCypressConfig(configFilePath, context);

  const namedInputs = getNamedInputs(projectRoot, context);

  const baseTargetConfig: TargetConfiguration<CypressExecutorOptions> = {
    executor: '@nx/cypress:cypress',
    options: {
      cypressConfig: configFilePath,
    },
  };

  const targets: Record<
    string,
    TargetConfiguration<CypressExecutorOptions>
  > = {};

  if ('e2e' in cypressConfig) {
    const e2eTargetDefaults = readTargetDefaultsForTarget(
      options.targetName,
      context.nxJsonConfiguration.targetDefaults,
      '@nx/cypress:cypress'
    );

    targets[options.targetName] = {
      ...baseTargetConfig,
      options: {
        ...baseTargetConfig.options,
        testingType: 'e2e',
      },
    };

    if (e2eTargetDefaults?.cache === undefined) {
      targets[options.targetName].cache = true;
    }

    if (e2eTargetDefaults?.inputs === undefined) {
      targets[options.targetName].inputs =
        'production' in namedInputs
          ? ['default', '^production']
          : ['default', '^default'];
    }

    if (e2eTargetDefaults?.outputs === undefined) {
      targets[options.targetName].outputs = getOutputs(
        projectRoot,
        cypressConfig,
        'e2e'
      );
    }

    const cypressEnv = {
      ...cypressConfig.env,
      ...cypressConfig.e2e?.env,
    };

    const devServerTargets: Record<string, string> =
      cypressEnv?.devServerTargets;

    if (devServerTargets?.default) {
      targets[options.targetName].options.devServerTarget =
        devServerTargets.default;
      delete devServerTargets.default;
    }

    if (Object.keys(devServerTargets ?? {}).length > 0) {
      targets[options.targetName].configurations ??= {};
      for (const [configuration, devServerTarget] of Object.entries(
        devServerTargets ?? {}
      )) {
        targets[options.targetName].configurations[configuration] = {
          devServerTarget,
        };
      }
    }

    const ciDevServerTarget: string = cypressEnv?.ciDevServerTarget;
    if (ciDevServerTarget) {
      const specPatterns = Array.isArray(cypressConfig.e2e.specPattern)
        ? cypressConfig.e2e.specPattern.map((p) => join(projectRoot, p))
        : [join(projectRoot, cypressConfig.e2e.specPattern)];

      const excludeSpecPatterns: string[] = !cypressConfig.e2e
        .excludeSpecPattern
        ? cypressConfig.e2e.excludeSpecPattern
        : Array.isArray(cypressConfig.e2e.excludeSpecPattern)
        ? cypressConfig.e2e.excludeSpecPattern.map((p) => join(projectRoot, p))
        : [join(projectRoot, cypressConfig.e2e.excludeSpecPattern)];
      const specFiles = globWithWorkspaceContext(
        context.workspaceRoot,
        specPatterns,
        excludeSpecPatterns
      );

      const dependsOn: TargetConfiguration['dependsOn'] = [];
      const outputs = getOutputs(projectRoot, cypressConfig, 'e2e');
      const inputs =
        'production' in namedInputs
          ? ['default', '^production']
          : ['default', '^default'];
      for (const file of specFiles) {
        const targetName = options.ciTargetName + '--' + file;
        targets[targetName] = {
          ...targets[options.targetName],
          outputs,
          inputs,
          cache: true,
          configurations: undefined,
          options: {
            ...targets[options.targetName].options,
            devServerTarget: ciDevServerTarget,
            spec: file,
          },
        };
        dependsOn.push({
          target: targetName,
          projects: 'self',
          params: 'forward',
        });
      }
      targets[options.ciTargetName] ??= {};

      targets[options.ciTargetName] = {
        executor: 'nx:noop',
        cache: true,
        inputs,
        outputs,
        dependsOn,
      };
    }
  }

  if ('component' in cypressConfig) {
    const componentTestingTargetDefaults = readTargetDefaultsForTarget(
      options.componentTestingTargetName,
      context.nxJsonConfiguration.targetDefaults,
      '@nx/cypress:cypress'
    );

    // This will not override the e2e target if it is the same
    targets[options.componentTestingTargetName] ??= {
      ...baseTargetConfig,
      options: {
        ...baseTargetConfig.options,
        testingType: 'component',
      },
    };

    if (componentTestingTargetDefaults?.cache === undefined) {
      targets[options.componentTestingTargetName].cache = true;
    }

    if (componentTestingTargetDefaults?.inputs === undefined) {
      targets[options.componentTestingTargetName].inputs =
        'production' in namedInputs
          ? ['default', '^production']
          : ['default', '^default'];
    }

    if (componentTestingTargetDefaults?.outputs === undefined) {
      targets[options.componentTestingTargetName].outputs = getOutputs(
        projectRoot,
        cypressConfig,
        'component'
      );
    }
  }

  return targets;
}

function getCypressConfig(
  configFilePath: string,
  context: CreateNodesContext
): any {
  const resolvedPath = join(context.workspaceRoot, configFilePath);

  let module: any;
  if (['.ts', '.mts', '.cts'].includes(extname(configFilePath))) {
    const tsConfigPath = getRootTsConfigPath();

    if (tsConfigPath) {
      const unregisterTsProject = registerTsProject(tsConfigPath);
      try {
        module = require(resolvedPath);
      } finally {
        unregisterTsProject();
      }
    } else {
      module = require(resolvedPath);
    }
  } else {
    module = require(resolvedPath);
  }
  return module.default ?? module;
}

function normalizeOptions(options: CypressPluginOptions): CypressPluginOptions {
  options ??= {};
  options.targetName ??= 'e2e';
  options.componentTestingTargetName ??= 'component-test';
  options.ciTargetName ??= 'e2e-ci';
  return options;
}
