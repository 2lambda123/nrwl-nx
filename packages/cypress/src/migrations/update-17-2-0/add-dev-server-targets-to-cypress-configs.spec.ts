import { createTree } from '@nx/devkit/testing';
import {
  addProjectConfiguration as _addProjectConfiguration,
  ProjectGraph,
  Tree,
} from '@nx/devkit';
import update from './add-dev-server-targets-to-cypress-configs';

let projectGraph: ProjectGraph;
jest.mock('@nx/devkit', () => ({
  ...jest.requireActual<any>('@nx/devkit'),
  createProjectGraphAsync: jest.fn().mockImplementation(async () => {
    return projectGraph;
  }),
}));

function addProjectConfiguration(tree, name, project) {
  _addProjectConfiguration(tree, name, project);
  projectGraph.nodes[name] = {
    name: name,
    type: 'lib',
    data: {
      root: project.root,
      targets: project.targets,
    },
  };
}

describe('add-dev-server-targets-to-cypress-configs migration', () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTree();

    tree.write(
      'e2e/cypress.config.ts',
      `
      import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nx/cypress/plugins/cypress-preset';

export default defineConfig({
  e2e: {
    ...nxE2EPreset(__filename),
  },
});
    `
    );

    projectGraph = {
      nodes: {},
      dependencies: {},
    };
  });

  it('should add devServerTargets to cypress.config.ts', async () => {
    addProjectConfiguration(tree, 'e2e', {
      root: 'e2e',
      targets: {
        e2e: {
          executor: '@nx/cypress:cypress',
          options: {
            cypressConfig: 'e2e/cypress.config.ts',
            devServerTarget: 'my-app:serve',
          },
        },
      },
    });

    await update(tree);

    expect(tree.read('e2e/cypress.config.ts', 'utf-8')).toMatchSnapshot();
  });

  it('should add dev server targets for default, production, and ci', async () => {
    addProjectConfiguration(tree, 'e2e', {
      root: 'e2e',
      targets: {
        e2e: {
          executor: '@nx/cypress:cypress',
          options: {
            cypressConfig: 'e2e/cypress.config.ts',
            devServerTarget: 'my-app:serve',
          },
          configurations: {
            production: {
              devServerTarget: 'my-app:serve:production',
            },
            ci: {
              devServerTarget: 'my-app:serve-static',
            },
          },
        },
      },
    });
    await update(tree);

    expect(tree.read('e2e/cypress.config.ts', 'utf-8')).toMatchSnapshot();
  });

  it('should update existing options with dev server targets for default, production, and ci', async () => {
    tree.write(
      'e2e/cypress.config.ts',
      `
      import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nx/cypress/plugins/cypress-preset';

export default defineConfig({
  e2e: {
    ...nxE2EPreset(__filename, { bundler: 'vite' }),
  },
});
    `
    );
    addProjectConfiguration(tree, 'e2e', {
      root: 'e2e',
      targets: {
        e2e: {
          executor: '@nx/cypress:cypress',
          options: {
            cypressConfig: 'e2e/cypress.config.ts',
            devServerTarget: 'my-app:serve',
          },
          configurations: {
            production: {
              devServerTarget: 'my-app:serve:production',
            },
            ci: {
              devServerTarget: 'my-app:serve-static',
            },
          },
        },
      },
    });
    await update(tree);

    expect(tree.read('e2e/cypress.config.ts', 'utf-8')).toMatchSnapshot();
  });

  it('should not add nx metadata for if there are none to add', async () => {
    addProjectConfiguration(tree, 'e2e', {
      root: 'e2e',
      targets: {
        e2e: {
          executor: '@nx/cypress:cypress',
          options: {},
        },
      },
    });
    await update(tree);

    expect(tree.read('e2e/cypress.config.ts', 'utf-8')).toMatchSnapshot();
  });
});
