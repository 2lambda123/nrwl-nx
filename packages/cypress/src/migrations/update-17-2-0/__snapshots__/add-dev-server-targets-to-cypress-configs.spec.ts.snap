// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-dev-server-targets-to-cypress-configs migration should add dev server targets for default, production, and ci 1`] = `
"import { defineConfig } from "cypress";
import { nxE2EPreset } from "@nx/cypress/plugins/cypress-preset";

export default defineConfig({
  e2e: {
    ...nxE2EPreset(__filename, {
      devServerTargets: {
        default: "my-app:serve",
        production: "my-app:serve:production",
      },
      ciDevServerTarget: "my-app:serve-static",
    }),
  },
});
"
`;

exports[`add-dev-server-targets-to-cypress-configs migration should add devServerTargets to cypress.config.ts 1`] = `
"import { defineConfig } from "cypress";
import { nxE2EPreset } from "@nx/cypress/plugins/cypress-preset";

export default defineConfig({
  e2e: {
    ...nxE2EPreset(__filename, {
      devServerTargets: { default: "my-app:serve" },
    }),
  },
});
"
`;

exports[`add-dev-server-targets-to-cypress-configs migration should not add nx metadata for if there are none to add 1`] = `
"import { defineConfig } from "cypress";
import { nxE2EPreset } from "@nx/cypress/plugins/cypress-preset";

export default defineConfig({
  e2e: {
    ...nxE2EPreset(__filename),
  },
});
"
`;

exports[`add-dev-server-targets-to-cypress-configs migration should update existing options with dev server targets for default, production, and ci 1`] = `
"import { defineConfig } from "cypress";
import { nxE2EPreset } from "@nx/cypress/plugins/cypress-preset";

export default defineConfig({
  e2e: {
    ...nxE2EPreset(__filename, {
      bundler: "vite",
      devServerTargets: {
        default: "my-app:serve",
        production: "my-app:serve:production",
      },
      ciDevServerTarget: "my-app:serve-static",
    }),
  },
});
"
`;
